src/traces/scattersmith/hover.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+var scatterHover = require('../scatter/hover');
+
+function hoverPoints(pointData, xval, yval, hovermode) {
+    var scatterPointData = scatterHover(pointData, xval, yval, hovermode);
+    if(!scatterPointData || scatterPointData[0].index === false) return;
+
+    var newPointData = scatterPointData[0];
+
+    // hovering on fill case
+    if(newPointData.index === undefined) {
+        return scatterPointData;
+    }
+
+    var subplot = pointData.subplot;
+    var cdi = newPointData.cd[newPointData.index];
+    var trace = newPointData.trace;
+
+    if(!subplot.isPtInside(cdi)) return;
+
+    newPointData.xLabelVal = undefined;
+    newPointData.yLabelVal = undefined;
+    makeHoverPointText(cdi, trace, subplot, newPointData);
+    newPointData.hovertemplate = trace.hovertemplate;
+    return scatterPointData;
+}
+
+function makeHoverPointText(cdi, trace, subplot, pointData) {
+    var radialAxis = subplot.radialAxis;
+    var angularAxis = subplot.angularAxis;
+    radialAxis._hovertitle = 're';
+    angularAxis._hovertitle = 'im';
+
+    var fullLayout = {};
+    fullLayout[trace.subplot] = {_subplot: subplot};
+    var labels = trace._module.formatLabels(cdi, trace, fullLayout);
+    pointData.rLabel = labels.rLabel;
+    pointData.thetaLabel = labels.thetaLabel;
+
+    if(!trace.hovertemplate) {
+        pointData.extraText = cdi.re + ' + ' + cdi.im + 'j';
+    }
```

Comment:
`hoverinfo` options should be handled here.
Here is a diff between `scatterpolar/hover.js` vs `scattersmith/hover.js`:

![smith-hover](https://user-images.githubusercontent.com/33888540/128379725-a9694be4-591f-4a2b-b3b5-7640b8ae4359.png)

Reply:
The options in scatterpolar don't really apply to the smith chart. In fact, I simplified this function because the smith chart hover text is not really configurable in the way that the polar chart's is.

Do you have a suggestion for what I should do here?

Reply:
Right now I see `hoverinfo` attribute in `scattersmith`.
```js
    hoverinfo: extendFlat({}, baseAttrs.hoverinfo, {
        flags: ['re', 'im', 'text', 'name']
    }),
```
So perhaps that should be dropped?

Reply:
9c9806e14236b1d9ce0b5a85730dfe742e927754